tools/actions_local_runner.py

```diff
@@ -117,176 +125,191 @@ async def shell_cmd(
+        return result.passed
 
+    async def run_helper(self) -> Optional[CommandResult]:
+        if self.files is not None:
+            relevant_files = self.filter_files(self.files)
+            if len(relevant_files) == 0:
+                # No files, do nothing
+                return CommandResult(passed=True, stdout="", stderr="")
 
-def header(name: str, passed: bool) -> None:
-    PASS = color(col.GREEN, "✓")
-    FAIL = color(col.RED, "x")
-    icon = PASS if passed else FAIL
-    print(f"{icon} {color(col.BLUE, name)}")
+            return await self.quick(relevant_files)
 
+        return await self.full()
 
-def get_flake_excludes() -> List[str]:
-    config = configparser.ConfigParser()
-    config.read(os.path.join(REPO_ROOT, ".flake8"))
+    def filter_ext(self, files: List[str], extensions: Set[str]) -> List[str]:
+        def passes(filename: str) -> bool:
+            return os.path.splitext(filename)[1] in extensions
 
-    excludes = re.split(r',\s*', config["flake8"]["exclude"].strip())
-    excludes = [e.strip() for e in excludes if e.strip() != ""]
-    return excludes
+        return [f for f in files if passes(f)]
 
+    def filter_files(self, files: List[str]) -> List[str]:
+        return files
 
-async def run_flake8(files: Optional[List[str]], quiet: bool) -> bool:
-    cmd = ["flake8"]
+    async def quick(self, files: List[str]) -> CommandResult:
+        raise NotImplementedError
 
-    excludes = get_flake_excludes()
+    async def full(self) -> Optional[CommandResult]:
+        raise NotImplementedError
 
-    def should_include(name: str) -> bool:
-        for exclude in excludes:
-            if fnmatch.fnmatch(name, pat=exclude):
-                return False
-            if name.startswith(exclude) or ("./" + name).startswith(exclude):
-                return False
-        return True
 
-    if files is not None:
-        files = [f for f in files if should_include(f)]
+class Flake8(Check):
+    name = "flake8"
 
-        if len(files) == 0:
-            print_results("flake8", True, [])
+    def filter_files(self, files: List[str]) -> List[str]:
+        config = configparser.ConfigParser()
+        config.read(os.path.join(REPO_ROOT, ".flake8"))
+
+        excludes = re.split(r",\s*", config["flake8"]["exclude"].strip())
+        excludes = [e.strip() for e in excludes if e.strip() != ""]
+
+        def should_include(name: str) -> bool:
+            for exclude in excludes:
+                if fnmatch.fnmatch(name, pat=exclude):
+                    return False
+                if name.startswith(exclude) or f"./{name}".startswith(exclude):
+                    return False
             return True
 
-        # Running quicklint, pass in an explicit list of files (unlike mypy,
-        # flake8 will still use .flake8 to filter this list by the 'exclude's
-        # in the config
-        cmd += files
+        files = self.filter_ext(files, {"py"})
```

Comment:
you need the `.`:
```suggestion
        files = self.filter_ext(files, {".py"})
```

Reply:
Nice catch, I thought the GitHub UI commit buttons would break ghstack but it looks like everything turned out fine

Reply:
oh weird! I also thought they would break `ghstack`, since I'm pretty sure I've seen that happen in the past... strange

Reply:
Just kidding it did break it (and silently too!), it shows the changes applied on the GitHub UI but `ghstack checkout` doesn't have them nor does the `ghimport`ed diff

Reply:
oh sad :cry: